### DISCUSSION TITLE
2.0 可以自定义生命周期吗

### DISCUSSION THREAD

**Discussion by RihuaChen** [alibaba/formily#1800] [Q&A 2.x] [answered]
2.0这边只有这个createEffectHook，能做到1.x那种$('onFocus', 'path')这样的吗，不然只能在一个方法里面写if去判断哪个触发了这个生命周期

**Comment by janryWang** [comment, marked as answer]
都用2.x了，就尽量少用自定义生命周期，直接reactive包装一下数据，到处都能响应

**Reply by RihuaChen** [reply]
可以举个例子吗，有点不太理解，比如在onFocus要抛出来数据，需要怎么操作一下呢 = =

**Reply by janryWang** [reply]
x-component-props传onFocus不就行了

**Reply by RihuaChen** [reply]
这个主要需要走配置化，不能用scope这样的方式才这样绕一圈来做，插值的都会被序列化掉，所以才这么做的

**Reply by janryWang** [reply]
为啥插值会被序列化掉，用表达式不行吗

**Reply by RihuaChen** [reply]
因为是从useFormSchema来的。。。
